thirdeye/thirdeye-frontend/app/utils/api/self-serve.js

```diff
@@ -142,20 +102,93 @@ const createNewDataset = '/onboard/create';
  */
 const setAlertActivationUrl = (detectionId, active) => `/yaml/activation/${detectionId}?active=${active}`;
 
+/**
+ * GET autocomplete request for alert by name
+ * @param {String} functionName: alert name
+ * @see {@link https://tinyurl.com/ybelagey|class DataResource}
+ */
+const alertByName = functionName => `/data/autocomplete/detection?name=${encodeURIComponent(functionName)}`;
+
+/**
+ * GET autocomplete request for application
+ * @param {String} app: application name
+ */
+const application = app => `/data/autocomplete/application?name=${encodeURIComponent(app)}`;
+
+/**
+ * GET autocomplete request for dataset
+ * @param {String} type: dataset name
+ */
+const dataset = name => `/data/autocomplete/dataset?name=${encodeURIComponent(name)}`;
+
+/**
+ * GET autocomplete request for detection type
+ * @param {String} type: detection type
+ */
+const detectionType = type => `/data/autocomplete/ruleType?name=${encodeURIComponent(type)}`;
+
+/**
+ * GET autocomplete request for metric data by name
+ * @param {String} metricName: metric name
+ * @see {@link https://tinyurl.com/y7hhzm33|class DataResource}
+ */
+const metric = metricName => `/data/autocomplete/metric?name=${encodeURIComponent(metricName)}`;
+
+/**
+ * GET autocomplete request for alerty owner
+ * @param {String} name: owner name
+ */
+const owner = name => `/data/autocomplete/detection-createdby?name=${encodeURIComponent(name)}`;
+
+/**
+ * GET autocomplete request for subscription group by name
+ * @param {String} name: group name
+ */
+const subscriptionGroup = name => `/data/autocomplete/subscription?name=${encodeURIComponent(name)}`;
+
+/**
+ * GET alerts, filtered and paginated according to params
+ * @param {Object} params: params to query alerts by
+ */
+const getPaginatedAlertsUrl = params => {
+  const paramKeys = Object.keys(params || {});
+  let url = '/alerts';
+  if (paramKeys.length === 0) {
+    return url;
+  }
+  url += '?';
+  let index = 0;
+  paramKeys.forEach(key => {
+    const paramString = `${key}=${encodeURIComponent(params[key])}`;
```

Comment:
It seems the request URL for when there are multiple values is incorrect. it should be `/alerts?limit=10&offset=0&application=app1&application=app2` not `/alerts?limit=10&offset=0&application=app1%2Capp2`. Otherwise, it won't get the right result. Could you double check?